### ISSUE TITLE
WG Revision: MathML 4

### ISSUE THREAD

**Issue by NSoiffer** [w3ctag/design-reviews#1205]
### Specification

https://www.w3.org/TR/mathml4

### Explainer

https://w3c.github.io/mathml-docs/intent-explainer/

### Links

- The WG's request for this TAG review: https://lists.w3.org/Archives/Public/www-math/2025Dec/0007.html
- TAG review of the previous version of this specification, if any:  MathML 3 (CR in 2009) apparently predates TAG review, I couldn't find anything.  MathML Core's review is at: https://github.com/w3ctag/design-reviews/issues/438.
- A description of what has changed since our previous review:
  There are two major changes:
  1. Alignment with [MathML Core](https://www.w3.org/TR/mathml-core/), which was reviewed before it went into CR.
  2. The addition of "[author intent](https://www.w3.org/TR/mathml4/#mixing_intent)". These are attributes that can be used by authors to disambiguate the speech for a notation. The goal is improved accessibility. This is the significant change that should be reviewed.

Feature 1:
- Previous early design review, if any:  None. There were discussions around 2022 with the ARIA WG that led to the conclusions in the explainer about ARIA not being an appropriate solution.
- An introduction to the feature, aimed at unfamiliar audiences:  see the early parts of [the explainer](https://w3c.github.io/mathml-docs/intent-explainer/).
- A description of the problems that end-users were facing before this proposal: https://w3c.github.io/mathml-docs/intent-explainer/#semantic-reading-considerations
- Alternatives considered: https://w3c.github.io/mathml-docs/intent-explainer/#leveraging-existing-technology
- Examples of how to use the proposal to solve the end-users' problems:  [a few simple examples are shown in the explainer](https://w3c.github.io/mathml-docs/intent-explainer/#a-mathml-specific-solution). A much [larger set of examples can be found here](https://w3c.github.io/mathml-docs/intent-examples/).
Note: the speech in the larger set of examples comes from MathCAT, which is used by the Windows-based screen readers JAWS and NVDA among others. MathCAT does far more inference of speech than other accessibility solutions, so the difference between the default reading and the intent reading is much smaller than is the case in other screen readers. For example, $P^n_k$ is not likely to be read as "k permutations of n" by most other MathML-to-speech solutions. They are more likely to read it like "p subscript k superscript n end superscript" or (incorrectly) as "p sub k to the nth power".
- What do the end-users experience with this proposal: see above answer with links to examples of speech.
- User research you did to validate the problem and/or design, if any?
  As mentioned in [the explainer](https://w3c.github.io/mathml-docs/intent-explainer/#semantic-reading-considerations), there are no studies showing semantic speech is better than literal speech. But it is typically shorter (hence, it uses less working memory) and it corresponds to what is spoken in the classroom and elsewhere. So it is generally believed to be preferable. Validation of the design was done by examining [many examples](https://w3c.github.io/mathml-docs/intent-examples/) along with making sure it could both be generated by authoring tools and consumed by assistive technology. In particular, the addition of intent `properties` solved a long-standing problem of how to speak [some tabular layouts](https://w3c.github.io/mathml-docs/intent-examples/#id-0-34f18cf999cd8a7d05e035ab7c3ca389) well. These are often represented as tables to align parts of an equation such as an equals sign and otherwise get spoken with "row 1, col 1: ..., col2 ..." when in fact they are not tabular.
- Web Platform Tests:  MathML Core has Web Platforms Tests. There are no platform independent tests for `intent` because the spec intentionally does not require a specific way to speak the math. Different speech is appropriate for different audiences (see [the explainer](https://w3c.github.io/mathml-docs/intent-explainer/#examples)).

### The specification

- [x] Follows the [Web Platform Design Principles](https://www.w3.org/TR/design-principles/).
- [x] Includes Security and Privacy Considerations sections based on answers to the [Security/Privacy Questionnaire](https://www.w3.org/TR/security-privacy-questionnaire/).

### Where and by whom is the work is being done?

- GitHub repo:  https://github.com/w3c/mathml/issues
- Primary contacts:
  - Neil Soiffer (@nsoiffer), Talking Cat Software, Math WG Co-chair
  - Brian Kardell (@bkardell), Igalia, Math WG Co-chair
- W3C Math WG
- This work is being funded by: None. Volunteer-based with each authoring and AT vendor self-funding their implementation
- Primary standards group developing this feature: Math WG
- Incubation and standards groups that have discussed the design:
  - sorry, I can't find the discussion links with ARIA and internationalization groups

### Feedback so far

- Active horizontal reviews:
- Privacy: https://github.com/w3cping/privacy-request/issues/197
- Security: https://github.com/w3c/security-request/issues/121
- Accessibility: https://github.com/w3c/a11y-request/issues/151
- Internationalization: https://github.com/w3c/i18n-request/issues/299

- Multi-stakeholder feedback:
  - Browsers only pass the `intent` and `arg` attributes through, they do not process them and so no comments.
  - The following all are happy with and contributed in the development of `intent`:
    - MathCAT is used by many Windows-based assistive technology (JAWS, NVDA, Dolphin EasyReader, Kurzweil 3000, ...) and it has implemented most of the intent specification
    - The LaTeX project, which maintains TeX/LaTeX has implemented some intent generation and have added a macro to generate intent. They plan to incorporate that macro into many of the LaTeX packages
     - The Wikimedia foundation has incorporated intent into one of its MathML editors
- Adobe Acrobat/Reader, Foxit Editor/Reader, and Firefox PDF's viewer all support MathML in PDF and pass along the intent attributes.
- Major unresolved issues with or opposition to this specification: none, but the Math WG continues to refine the core concept list and properties based on usage and feedback. In submitting this review, I became aware of the [registries](https://www.w3.org/policies/process/#registries) and the Math WG will look into using that for `intent` tables.
- Status/issue trackers for implementations:  none yet

### You should also know that...

As a reminder, the only new addition besides aligning with MathML core, is [`intent`](https://www.w3.org/TR/mathml4/#mixing_intent), which is a small part of the spec.

The Math WG has started the re-chartering process. Our charter ends in April. It would be nice, but not essential, to start the new WG (assuming the group is re-chartered) with the reviews in hand so we can move to CR and move that stage along.

**Comment by jyasskin** [comment]
Thanks for sending us this review. We were concerned that you might be re-inventing semantic MathML, but after discussion, we're satisfied that this syntax is more helpful to practical assistive technology tools. We appreciate that you're coordinating with the ARIA WG to ensure that this fits into the rest of the platform. We noticed that you have separate documents for [core](https://w3c.github.io/mathml-docs/intent-core-concepts/) and [open](https://w3c.github.io/mathml-docs/intent-open-concepts/) lists of concepts. We suspect that these should be [W3C Registries](https://www.w3.org/policies/process/#registries) instead of plain documents.
